### ISSUE TITLE
Upgrade Promise dependency before Xcode 13 is released

### ISSUE THREAD

**Issue by ptxmac** [firebase/firebase-ios-sdk#8334]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13 beta 2
  * Firebase SDK version: 8.2.0
  * Installation method: `CocoaPods`
  * Firebase Component: Core

### [REQUIRED] Step 2: Describe the problem

The current version of Xcode 13 (beta 2) have a change to the swift compiler which makes `await` into a keyword. It seems this change is not guarded be a feature flag, so anything compiled with Xcode 13 (even if not targeting new versions) will not be able to use `await` as a function name (See [0296-async](https://github.com/apple/swift-evolution/blob/main/proposals/0296-async-await.md#source-compatibility) ).

This conflicts with [google/promise](https://github.com/google/promises) which Firebase depends on (through `GoogleUtilities/Environment`).

`Google/Promises` have been updated with a fix that renames `await`, but required a major release (v2.0.0).

Since Firebase is still depending on the old version, any project that use both Firebase and Promise will not be able to compile with Xcode 13 since there's no way to allow two versions of Promise in one project.

#### Steps to reproduce:

1. Create a project that Use both Firebase and Promise and compile it with Xcode 13
2. Trying to use the old Promise will result in compile errors
3. Updating to use Promise 2 is not possible because of firebase dependencies:
```
        FirebaseCore (~> 8.0) was resolved to 8.2.0, which depends on
          GoogleUtilities/Environment (~> 7.4) was resolved to 7.4.3, which depends on
            PromisesObjC (~> 1.2)
```

**Comment by ptxmac** [comment]
Quick update, did some investigation and found the following components needs update:
* GoogleUtilities/Environment
* GoogleDataTransport
* FirebaseCrashlytics
* FirebaseAppCheck
* FirebaseInstallations

**Comment by paulb777** [comment]
@ptxmac Thanks for the report!

It looks like it should be safe to update our dependencies to allow either Promises 1.x or 2.x, since at first glance I don't see any usages of the `await` API.

We could change our podspecs and Package.swift along the lines of https://github.com/CocoaPods/Specs/blob/master/Specs/5/9/a/Google-Mobile-Ads-SDK/8.7.0/Google-Mobile-Ads-SDK.podspec.json#L5

**Comment by denandreychuk** [comment]
I have similar problem with FirebaseFirestoreSwift pod

[!] CocoaPods could not find compatible versions for pod "FirebaseFirestore":
  In snapshot (Podfile.lock):
    FirebaseFirestore (= 8.2.0)

  In Podfile:
    FirebaseFirestoreSwift was resolved to 0.4.0, which depends on
      FirebaseFirestore (>= 1.6.1, ~> 1.6)

    Pimine/Firestore (from `https://github.com/pimine/piminesdk`) was resolved to 0.6.0, which depends on
      FirebaseFirestore

**Comment by paulb777** [comment]
@denandreychuk You're possibly running into https://github.com/firebase/firebase-ios-sdk/issues/7195#issuecomment-748160262. If not, please open a separate issue.

**Comment by ptxmac** [comment]
Small update: I created my own forks of the different dependencies and upgraded `PromiseObjC` to version 2 in all of them and had no problems - everything compiles without issue (at for the part I needed in my project)

**Comment by paulb777** [comment]
@ptxmac Thanks for the confirmation! We plan to get the fix in the 8.4.0 release.

**Comment by paulb777** [comment]
See google/GoogleUtilities#47 and google/GoogleDataTransport#31
